projects/vdk-plugins/vdk-ipython/tests/test_plugin.py

```diff
@@ -18,21 +21,151 @@ def ip(session_ip):
+    assert ip.user_global_ns["VDK"] is not None
+    assert isinstance(ip.user_global_ns["VDK"], JobControl)
+    # first call
+    ip.get_ipython().run_cell("job_input = VDK.get_initialized_job_input()")
+    result_job_input = ip.get_ipython().getoutput("job_input")
+    assert ip.user_global_ns["job_input"] is not None
+    assert isinstance(ip.user_global_ns["job_input"], IJobInput)
+    # second call
+    ip.get_ipython().run_cell("job_input = VDK.get_initialized_job_input()")
     assert ip.user_global_ns["job_input"] is not None
     assert isinstance(ip.user_global_ns["job_input"], IJobInput)
-    assert ip.user_global_ns["job_input"].get_name() == "test"
+    assert result_job_input == ip.get_ipython().getoutput("job_input")
 
 
-def test_load_job_input_with_invalid_argument(ip):
-    with pytest.raises(
-        UsageError, match=r"unrecognized arguments: --invalid_arg=dummy"
-    ):
-        ip.run_line_magic(magic_name="reload_job_input", line="--invalid_arg=dummy")
+# uses the pytest tmpdir fixture - https://docs.pytest.org/en/6.2.x/tmpdir.html#the-tmpdir-fixture
+async def test_extension_with_ingestion_job(ip, tmpdir):
+    # set environmental variables via Jupyter notebook
+    job_dir = str(tmpdir) + "vdk-sqlite.db"
+    ip.get_ipython().run_cell("%env VDK_INGEST_METHOD_DEFAULT=sqlite")
+    ip.get_ipython().run_cell(f"%env VDK_SQLITE_FILE={job_dir}")
+    ip.get_ipython().run_cell("%env VDK_DB_DEFAULT_TYPE=SQLITE")
+
+    ip.run_line_magic(magic_name="reload_VDK", line="")
+
+    # get the job_input
+    ip.get_ipython().run_cell("job_input = VDK.get_initialized_job_input()")
+
+    # execute SQL using job_input
+    ip.get_ipython().run_cell(
+        'job_input.execute_query("DROP TABLE IF EXISTS rest_target_table;")'
+    )
+    ip.get_ipython().run_cell(
+        'job_input.execute_query("CREATE TABLE rest_target_table (userId, id, title, completed);")'
+    )
+
+    # check whether the table was created
+    # the output should look like this:
+    #  'userId    id    title    completed',
+    #  '--------  ----  -------  -----------'
+    assert "userId    id    title    completed" in (
+        ip.get_ipython().getoutput(
+            "! " "vdk " "sqlite-query -q 'SELECT * FROM rest_target_table'"
+        )
+    )
+    assert "--------  ----  -------  -----------" in (
+        ip.get_ipython().getoutput(
+            "! " "vdk " "sqlite-query -q 'SELECT * FROM rest_target_table'"
+        )
+    )
+
+    # get the data that is going to be ingested
+    ip.get_ipython().run_cell("import requests")
+    ip.get_ipython().run_cell(
+        'response = requests.get("https://jsonplaceholder.typicode.com/todos/1")'
+    )
+    ip.get_ipython().run_cell("payload = response.json()")
+
+    # send data for ingestion
+    ip.get_ipython().run_cell(
+        'job_input.send_object_for_ingestion(payload=payload,destination_table="rest_target_table")'
+    )
+
+    # wait until the table is populated
+    await asyncio.sleep(1)
```

Comment:
There is `time.sleep` to sleep . There's not need to add 3th party library just for that.

How sure you are that 1 second is enough ?  It's always better not to rely on timing but to verify explicitly (e.g wait till it's available)

Reply:
The vdk-sqlite ingestion tests seem to directly lookup the data upon ingestion, see https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-plugins/vdk-sqlite/tests/test_sqlite_plugin.py#L71

Reply:
> The vdk-sqlite ingestion tests seem to directly lookup the data upon ingestion, see https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-plugins/vdk-sqlite/tests/test_sqlite_plugin.py#L71

I used it because of the get_ipython.get_output function